partitioned-table: Correct the variable name for list partitioning (#5227)
pingcap/docs: partitioned-table.md

>
> List partitioning is an experimental feature. It is not recommended that you use it in the production environment.

Before creating a List partitioned table, you need to set the value of the session variable `tidb_enable_list_partition` to `ON`.

{{< copyable "sql" >}}

```sql
set @@session.tidb_enable_list_partition = ON
```

> Also `tidb_enable_table_partition` needs to be set to `ON`, but this is the default.

Reviewer: ```suggestion
Also, make sure that `tidb_enable_table_partition` is set to `ON`, which is the default setting.
```

Reviewer 2: Please address this comment.

REST LGTM
